### ISSUE TITLE
Naming/UncommunicativeBlockParamName and Naming/VariableName false positives with keyword arguments

### ISSUE THREAD

**Issue by dgollahon** [rubocop-hq/rubocop#5837]
I encountered an interesting case with these two naming cops. I was mapping over an array of hashes and my favorite way to do this in ruby is to use keyword args to destructure the data. I've included a trivialized example below. The naming cops complain because the original data I had did not conform to ruby's variable conventions, but these were not names that I personally introduced--I was merely ingesting external data.

This got me thinking--I don't think keyword block arguments should actually be considered as introducing a name. The name _must_ match a previously existing name or you get an `ArgumentError` for a missing keyword. Therefore I don't think these cops should police these arguments, but rather object to naming where names are introduced only (i.e. method definitions).

--------

## Expected behavior

```ruby
require 'json'

data =
  JSON.parse(<<~JSON, symbolize_names: true)
    [
      {
        "notCamelCase": "data",
        "otherData": null
      }
    ]
  JSON

data_i_care_about =
  data.map do |notCamelCase:, **|
    { better_name: notCamelCase }
  end
```

I expect no violation to be raised.

## Actual behavior

![image](https://user-images.githubusercontent.com/4211073/39403001-f6be752c-4b23-11e8-87a5-3689c6786dd3.png)

Both the mentioned cops report a violation.

## Steps to reproduce the problem

Run `rubocop` on the above code sample.

## RuboCop version

```
$ rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by dgollahon** [comment]
This still bugs me when I'm ingesting data with keys that I didn't name. Since keyword argument binding (usually*) means you're not **"introducing" a name, but are accessing a name (vaguely related: there's no flag for using `:fooBar` in `hash.fetch(:fooBar)`) so it seems like it should not flag by default or be an option to turn off.

*Actually, i guess you _are_ "introducing" a name in some cases like in a lambda or a dynamic method definition, but I think that's less common.
```ruby
lambda do |fooBar:|
	fooBar
end
```

**Obviously the name is still (re-)introduced to the scope, but I still think it's cleaner than having to do a `fetch` and rename separately in most cases.
